engine/engine.go

```diff
@@ -556,49 +562,121 @@ func (e *Engine) publishAdvForIndex(ctx context.Context, contextID []byte, md me
 	return e.Publish(ctx, adv)
 }
 
-func (e *Engine) putKeyCidMap(ctx context.Context, contextID []byte, c cid.Cid) error {
+func (e *Engine) keyToCidKey(provider string, contextID []byte) datastore.Key {
```

Comment:
Would it make sense to take `peer.ID` wherever possible instead of `string` type?

Reply:
Fixed